frontend/dashboard/pages/PageLayout/PageLayout.test.tsx

```diff
@@ -64,10 +63,45 @@ describe('PageLayout', () => {
 
   test('should redirect to root if user does not have access to selected context', async () => {
     (useParams as jest.Mock).mockReturnValue({
-      selectedContext: 'test',
+      selectedContext: 'testinvalidcontext',
     });
     renderWithMockServices();
     expect(mockedNavigate).toHaveBeenCalledTimes(1);
-    expect(mockedNavigate).toHaveBeenCalledWith(DASHBOARD_ROOT_ROUTE);
+    expect(mockedNavigate).toHaveBeenCalledWith(SelectedContextType.Self, expect.anything());
+  });
+
+  test('should redirect to self context if none is defined', async () => {
+    renderWithMockServices();
+    expect(mockedNavigate).toHaveBeenCalledTimes(1);
+    expect(mockedNavigate).toHaveBeenCalledWith(SelectedContextType.Self, expect.anything());
+  });
+
+  test.each([
+    ['"self"', 'self'],
+    ['"all"', 'all'],
+    ['"ttd"', 'ttd'],
+  ])(
+    'should redirect to last selected context if none is selected, selected: %s',
+    async (context, navparam) => {
+      (useParams as jest.Mock).mockReturnValue({
+        selectedContext: SelectedContextType.None,
+      });
+      sessionStorage.setItem('dashboard::selectedContext', context);
+      renderWithMockServices();
+      expect(mockedNavigate).toHaveBeenCalledWith(navparam, expect.anything());
+    },
+  );
+
+  test('should redirect to self if user does not have access to session stored context', async () => {
+    (useParams as jest.Mock).mockReturnValue({});
+    sessionStorage.setItem('dashboard::selectedContext', '"testinvalidcontext"');
+    renderWithMockServices();
+    expect(mockedNavigate).toHaveBeenCalledTimes(1);
+    expect(mockedNavigate).toHaveBeenCalledWith(SelectedContextType.Self, expect.anything());
+  });
+
+  afterEach(() => {
+    sessionStorage.clear();
+    mockedNavigate.mockReset();
```

Comment:
We usually place afterEach, beforeEach, etc. at the top before all the tests, right after the describe. ;)